hyper/http11/connection.py

```diff
@@ -137,6 +144,33 @@ def connect(self):
 
         return
 
+    @classmethod
+    def _create_tunnel(cls, proxy_host, proxy_port, target_host, target_port):
+        """
+        Sends CONNECT method to a proxy and returns a socket with established
+        connection to the target.
+
+        :returns: socket
+        """
+        conn = cls(proxy_host, proxy_port)
+        conn.connect()
+        conn._send_headers(to_bytestring('CONNECT'),
+                           to_bytestring('%s:%d' % (target_host, target_port)),
+                           # TODO proxy-authorization headers
+                           headers=HTTPHeaderMap())
+
+        response = None
+        while response is None:
+            # 'encourage' the socket to receive data.
+            conn._sock.fill()
+            response = conn.parser.parse_response(conn._sock.buffer)
```

Comment:
Yes, the reason is that `HTTP11Response` `assert`s on presence of some headers like `Connection`/`Content-length` which are absent in the proxy response to a CONNECT. 

Maybe I should add a comment here explaining this. Or should we adapt the `HTTP11Response` to accept responses without these headers?